feat(number-format): Add duration formatter with colon notation

gerbermichi opened the pull request:
<!---
Please write the PR title following the conventions at https://www.conventionalcommits.org/en/v1.0.0/
Example:
fix(dashboard): load charts correctly
-->

### SUMMARY
Introduces a new formatter, DURATION_COL, for durations presented in colon notation (e.g., 0:10.5). 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<img width="398" alt="image" src="https://github.com/user-attachments/assets/0970461d-616f-41c2-96ce-d4d3c85e0fb0">


### TESTING INSTRUCTIONS
1. Create a chart with a duration (in ms) column 
2. Change the number formatter to `Duration in ms (10500 => 0:10.5)`

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

villebro commented on superset-frontend/packages/superset-ui-core/src/number-format/factories/createTimeDurationFormatter.ts:
```diff
@@ -0,0 +1,43 @@
+
+import NumberFormatter from '../NumberFormatter';
+
+export default function createTimeDurationFormatter(
+  config: {
+    description?: string;
+    id?: string;
+    label?: string;
+    multiplier?: number;
+  } = {},
+) {
+  const { description, id, label } = config;
+
+  return new NumberFormatter({
+    description,
+    formatFunc: value => {
+      const v = Math.abs(Math.round(value));
+      const h = Math.floor(v / 60);
+      const m = v % 60;
+      return `${h}:${String(m).padStart(2, '0')}`;
+    },
+    id: id ?? 'duration_format',
+    label: label ?? `Duration formatter`,
+  });
+}
```
We already have a dependency on [pretty-ms](https://www.npmjs.com/package/pretty-ms) that achieves this and (see `colonNotation` option).

villebro commented on superset-frontend/src/setup/setupFormatters.ts:
```diff
@@ -78,6 +79,7 @@ export default function setupFormatters(
       'DURATION_SUB',
       createDurationFormatter({ formatSubMilliseconds: true }),
     )
+    .registerValue('TIME_DURATION', createTimeDurationFormatter())
```
Please check the `README.md` in the `number-format` directory for an example how to implement this. I believe this should do it:
```suggestion
    .registerValue('TIME_DURATION', createTimeDurationFormatter({ colonNotation: true }))
```

gerbermichi replied:
Thank you. This does not the same, it will format it in hh:mm:ss and I would like to create a formatter for hh:mm. However, I can add both, DURATION_COL and DURATION_COL_SHORT. What are your thoughts?

villebro reviewed (changes requested):
We don't need to create a new formatter for this, as the existing duration formatter already supports this functionality. Also, let's stick to milliseconds for now to keep this aligned with the existing duration formatters.

villebro commented:
FYI: I opened a PR to bump `pretty-ms` to pull in a critical fix for `colonNotation`, as that's what we should be using here: #30599

gerbermichi committed:
Add support for DURATION_COL formatter

Introduces a new formatter, DURATION_COL, for durations presented in colon notation (e.g., 0:10.5). This update enhances the flexibility of duration display formats within the application.
